templates/SilverStripe/CMS/Forms/SiteTreeURLSegmentField.ss

```diff
@@ -1,20 +1,31 @@
+<% if not $IsReadonly %>
```

Comment:
You don't need an if statement here. You could just have `SiteTreeURLSegmentField_Readonly.ss` file to handle the readonly bit.